# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2024/05
# 1) change AreTomo arg --outImod to value of environment variable WARPWORKER_ARETOMO_OUTIMOD or default to 1
# if unset
# 2) Allow to define additional AreTomo2 args by setting environment variable 
#    WARPWORKER_ARETOMO_EXTRAARGS
diff -ru warp-c8b254eb807bb7ae52f1b5bb77f0921fbe9e885e/WarpWorker/WarpWorker.cs warp-c8b254eb807bb7ae52f1b5bb77f0921fbe9e885e_outimod/WarpWorker/WarpWorker.cs
--- warp-c8b254eb807bb7ae52f1b5bb77f0921fbe9e885e/WarpWorker/WarpWorker.cs	2024-05-18 03:13:48.000000000 +0200
+++ warp-c8b254eb807bb7ae52f1b5bb77f0921fbe9e885e_outimod/WarpWorker/WarpWorker.cs	2024-05-21 10:49:49.012553734 +0200
@@ -352,8 +352,11 @@
                     string AlignZ = Options.AlignZ.ToString();
                     string NPatchesX = Options.NPatchesXY[0].ToString();
                     string NPatchesY = Options.NPatchesXY[1].ToString();
+                    string OutImod = (Environment.GetEnvironmentVariable("WARPWORKER_ARETOMO_OUTIMOD")!=null ) ? Environment.GetEnvironmentVariable("WARPWORKER_ARETOMO_OUTIMOD") : "1";
+                    string AretomoExtraArgs = (Environment.GetEnvironmentVariable("WARPWORKER_ARETOMO_EXTRAAARGS")!=null ) ? Environment.GetEnvironmentVariable("WARPWORKER_ARETOMO_EXTRAARGS") : "";
+
+                    string Arguments = $"-InMrc {StackPath} -AngFile {AnglePath} -VolZ 0 -OutBin 0 -TiltAxis {Axis} -AlignZ {AlignZ} -TiltCor 1 -OutImod {OutImod} -DarkTol 0 -OutMrc {OutStack} -Gpu {DeviceID} -Patch {NPatchesX} {NPatchesY} {AretomoExtraArgs}";
 
-                    string Arguments = $"-InMrc {StackPath} -AngFile {AnglePath} -VolZ 0 -OutBin 0 -TiltAxis {Axis} -AlignZ {AlignZ} -TiltCor 1 -OutImod 1 -DarkTol 0 -OutMrc {OutStack} -Gpu {DeviceID} -Patch {NPatchesX} {NPatchesY}";
 
                     Console.WriteLine($"Executing {Options.Executable} in {StackDir} with arguments: {Arguments}");
 
